src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -158,6 +153,76 @@ public static class CreateMutationBuilder
             return input;
         }
 
+        /// <summary>
+        /// Generate the GraphQL input type from an object type for non-relational database.
+        /// </summary>
+        /// <param name="inputs">Reference table of all known input types.</param>
+        /// <param name="objectTypeDefinitionNode">GraphQL object to generate the input type for.</param>
+        /// <param name="name">Name of the GraphQL object type.</param>
+        /// <param name="definitions">All named GraphQL items in the schema (objects, enums, scalars, etc.)</param>
+        /// <param name="databaseType">Database type of the non-relational database to generate input type for.</param>
+        /// <param name="entities">Runtime config information.</param>
+        /// <returns>A GraphQL input type with all expected fields mapped as GraphQL inputs.</returns>
+        private static InputObjectTypeDefinitionNode GenerateCreateInputTypeForNonRelationalDb(
+            Dictionary<NameNode, InputObjectTypeDefinitionNode> inputs,
+            ObjectTypeDefinitionNode objectTypeDefinitionNode,
+            NameNode name,
+            IEnumerable<HotChocolate.Language.IHasName> definitions,
+            DatabaseType databaseType,
+            RuntimeEntities entities)
+        {
+            NameNode inputName = GenerateInputTypeName(name.Value);
```

Comment:
What was the need to add a separate function for NoSql dbs with duplicate code?

Reply:
There was a lot of non-overlapping code between cosmos and other relational dbs in terms of input type creation. This was eventually leading to a lot of `IF/ELSE` conditionals, making the code hard to understand and maintain (Please refer to the method: `GenerateCreateInputTypeForRelationalDb` and you'll see how much significantly complex is the logic for relational dbs from cosmos db.) I felt keeping everything in the same function is going to be problematic for developers across the relational/non-relational dbs as well. 

Hence, I seperated the two, even though it might mean a small duplication of code. Since `CreateMutationBuilder` is a static class, I didn't look to do a refactor of creating overridden implementations.